plugins/vpp/ifplugin/interface_config.go

```diff
@@ -461,10 +479,17 @@ func (plugin *InterfaceConfigurator) ModifyVPPInterface(newConfig *intf.Interfac
 	oldConfig *intf.Interfaces_Interface) error {
 	plugin.log.Infof("Modifying Interface %v", newConfig.Name)
 
+	// Recreate pending Af-packet
 	if newConfig.Type == intf.InterfaceType_AF_PACKET_INTERFACE && plugin.afPacketConfigurator.IsPendingAfPacket(oldConfig) {
 		return plugin.recreateVPPInterface(newConfig, oldConfig, 0)
 	}
 
+	// Re-create cached VxLAN
+	if _, ok := plugin.vxlanMulticastCache[newConfig.Name]; ok {
```

Comment:
This is probably missing some if condition for checking interface type. What if some interface of non-vxlan type has name that would exist in this cache? Wouldn't that cause issues?

Reply:
very unlikely but right, better to handle it anyway